### PR TITLE
Add validateEntity to catalog client

### PR DESCRIPTION
**Pull request by taras**
## Hey, I just made a Pull Request!

`/validate-entity` endpoint has been available on the Catalog REST API for a while. It was never added to the catalog client; as a result, it's not well known that it is available. 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Seems that vale wants you to put backticks around validateEntity in the changeset.

File: packages/catalog-client/src/CatalogClient.ts [hunk 1]
```diff
@@ -353,6 +354,41 @@ export class CatalogClient implements CatalogApi {
...
+      },
+    );
+
+    if (response.ok) {
+      return {
+        valid: true,
+        errors: undefined,
+      };
+    }
+
```
Comment (freben) [inline packages/catalog-client/src/CatalogClient.ts:383]: This should probably check for `if (response.status !== 400) { throw await ResponseError.fromResponse(response); }`

File: packages/catalog-client/src/types/api.ts [hunk 2]
```diff
@@ -269,6 +270,16 @@ export type AddLocationResponse = {
...
 
+/**
+ * The response type for {@link CatalogClient.validateEntity}
+ *
+ * @public
+ */
+export type ValidateEntityResponse = {
+  valid: boolean;
+  errors?: SerializedError[];
+};
```
Comment (freben) [inline packages/catalog-client/src/types/api.ts:281]: is it worth making this one `{ valid: true } | { valid: false; errors: SerializedError[] }`?

**Review by freben — COMMENTED** [review]
Aight!

File: packages/catalog-client/src/types/api.ts (hunk 2 above)
Comment (taras) [reply packages/catalog-client/src/types/api.ts]: Good idea. Done.

File: packages/catalog-client/src/CatalogClient.ts (hunk 1 above)
Comment (taras) [reply packages/catalog-client/src/CatalogClient.ts:382]: Done.

**Comment by taras** [comment]
@freben the least fun part of this PR was having to update every test that used uses Catalog Client. It's not something that has to happen often because that API doesn't change often, but it seems like a lot of tedious work to update tests in packages that are unrelated to this change. 

What do you think about creating a test helper that would provide a function for creating the catalog client mock? That way, it would only need to be changed in one place whenever the API needs to be changed. Against, this doesn't happen, but non-the-less.

**Review by freben — APPROVED** [review]

**Comment by freben** [comment]
Yeah maybe. There needs to be some form of overall strategy to how we generally do this mocking, that doesn't have this cost when the interface changes. But maybe it's time to start thinking about versioning of APIs and clients too, because technically now the interface actually had a breakage (if you've made your entirely own catalog client) and that's not good either. Gonna have to start thinking about things like that in general

**Pull request merged**